starter-code/src/viking.js

```diff
@@ -1,11 +1,114 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+
+  attack() {
+    return this.strength;
+  }
+
+  receiveDamage(theDamage) {
+    this.health -= theDamage;
+  }
+}
+
+// var soldier = new Soldier(150, 300);
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0)
+      return `${this.name} has received ${damage} points of damage`;
+    if (this.health === 0) return `${this.name} has died in act of combat`;
+  }
+
+  //     // if the Viking is still alive, it should return "NAME has received DAMAGE points of damage"
+  //     // if the Viking dies, it should return "NAME has died in act of combat"
+
+  battleCry() {
+    return "Odin Owns You All!";
+  }
+}
+
+// var viking = new Viking("Harald", 150, 300);
 
 // Saxon
-class Saxon {}
+
+class Saxon extends Soldier {
+  constructor(health, strength) {
+    super(health, strength);
+  }
+  receiveDamage(theDamage) {
+    this.health -= theDamage;
+    if (this.health > 0)
+      return `A Saxon has received ${theDamage} points of damage`;
+    if (this.health === 0) return `A Saxon has died in combat`;
+    // if the Saxon is still alive, it should return "A Saxon has received DAMAGE points of damage"
+    // if the Saxon dies, it should return "A Saxon has died in combat"
+  }
+}
+
+// var saxon = new Saxon(60, 25)
 
 // War
-class War {}
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+  addViking(viking) {
+    this.vikingArmy.push(viking);
+  }
+  addSaxon(saxon) {
+    this.saxonArmy.push(saxon);
+  }
+  vikingAttack() {
+    const randDomSaxonIndex = Math.floor(Math.random() * this.saxonArmy.length);
+    const randDomVikingIndex = Math.floor(
+      Math.random() * this.vikingArmy.length
+    );
+    const oneSaxon = this.saxonArmy[randDomSaxonIndex];
+    const oneViking = this.vikingArmy[randDomVikingIndex];
+    if (oneSaxon.health === 0) return this.saxonArmy.pop();
```

Comment:
array.pop() will remove the last element of your array.
But here your dead soldier has a random index, so it's maybe not him who is remove from the array.
To that, either you always chose the last soldier of the array to fight and then remove it with array.pop()
Or you can use array.splice(your random index, 1) so it will only remove your dead soldier